specification/search/data-plane/Azure.Search/stable/2025-09-01/searchservice.json

```diff
@@ -0,0 +1,12104 @@
+        }
+      },
+      "description": "Contains configuration options specific to the exhaustive KNN algorithm used during querying, which will perform brute-force search across the entire vector index."
+    },
+    "ExhaustiveKnnParameters": {
+      "type": "object",
+      "properties": {
+        "metric": {
+          "$ref": "#/definitions/VectorSearchAlgorithmMetric",
+          "x-nullable": true,
+          "description": "The similarity metric to use for vector comparisons."
+        }
+      },
+      "description": "Contains the parameters specific to exhaustive KNN algorithm."
+    },
+    "VectorSearchAlgorithmMetric": {
+      "type": "string",
+      "enum": [
+        "cosine",
+        "euclidean",
+        "dotProduct",
+        "hamming"
+      ],
+      "x-ms-enum": {
+        "name": "VectorSearchAlgorithmMetric",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "cosine",
+            "name": "Cosine",
+            "description": "Measures the angle between vectors to quantify their similarity, disregarding magnitude. The smaller the angle, the closer the similarity."
+          },
+          {
+            "value": "euclidean",
+            "name": "Euclidean",
+            "description": "Computes the straight-line distance between vectors in a multi-dimensional space. The smaller the distance, the closer the similarity."
+          },
+          {
+            "value": "dotProduct",
+            "name": "DotProduct",
+            "description": "Calculates the sum of element-wise products to gauge alignment and magnitude similarity. The larger and more positive, the closer the similarity."
+          },
+          {
+            "value": "hamming",
+            "name": "Hamming",
+            "description": "Only applicable to bit-packed binary data types. Determines dissimilarity by counting differing positions in binary vectors. The fewer differences, the closer the similarity."
+          }
+        ]
+      },
+      "description": "The similarity metric to use for vector comparisons. It is recommended to choose the same similarity metric as the embedding model was trained on."
+    },
+    "VectorSearchAlgorithmKind": {
+      "type": "string",
+      "enum": [
+        "hnsw",
+        "exhaustiveKnn"
+      ],
+      "x-ms-enum": {
+        "name": "VectorSearchAlgorithmKind",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "hnsw",
+            "name": "Hnsw",
+            "description": "HNSW (Hierarchical Navigable Small World), a type of approximate nearest neighbors algorithm."
+          },
+          {
+            "value": "exhaustiveKnn",
+            "name": "ExhaustiveKnn",
+            "description": "Exhaustive KNN algorithm which will perform brute-force search."
+          }
+        ]
+      },
+      "description": "The algorithm used for indexing and querying."
+    },
+    "VectorSearchCompressionConfiguration": {
```

Comment:
@johanste and @mikekistler this model dropped properties `rerankWithOriginalVectors` and `defaultOversampling` as they were moved into `RescoringOptions` compared to 2024-07-01.

From an SDK backwards compatibility perspective, what's the policy here? Does this warrant a new major version break in the SDKs or will they need to workaround this on the SDK side?